### ISSUE TITLE
Add the DeepDDI model

### ISSUE THREAD

**Issue by benedekrozemberczki** [AstraZeneca/chemicalx#2]
Dear @hzcheney,

- Please read the paper first. It is [here](https://www.pnas.org/content/115/18/E4304).
- After that read the contributing guidelines.
-   If there is an existing open source version of the model please take a look.
-   ChemicalX is built on top of [PyTorch 1.10.](https://pytorch.org/) and [torchdrug](https://torchdrug.ai/).
- A similar model is `` which uses``  to generate drug representations. Take a look at the layer definition [here]().
- The library heavily builds on top on [torchdrug](https://torchdrug.ai/) and molecules in batches are PackedGraphs.
- There is already a model class under `./chemicalx/models/`
- Context features, drug level features and labels are all FloatTensors.
- Look at the examples and tests under `./examples/` and `./tests/`.
- Add auxiliary layers as you see fit - please document these, add tests and add these layers to the main readme.md if needed.
- Add typing to the initialisation and forward pass.
- Non data dependent hyperparameters should have default values. 
- Please add tests under `./tests/` and make sure that your model/layer is tested with real data.
- Write an example under `./examples/`. What is the AUC on the test set? Is it reasonable?

**Comment by hzcheney** [comment]
Hi!😊Is this repo welcome for contribution?

**Comment by benedekrozemberczki** [comment]
Hi @hzcheney,

We are architecting the data loaders in January 2022 and after that, we will have a board with outstanding features and issues. I will get back to you!

Thank you for your interest! We want to hit KDD 2022 Applied Track.

Benedek

**Comment by hzcheney** [comment]
That will be great! Good luck on your paper!

**Comment by benedekrozemberczki** [comment]
Hi @hzcheney ,

Are you interested in contributing?

**Comment by hzcheney** [comment]
@benedekrozemberczki Yeah, I will work into this.

**Comment by benedekrozemberczki** [comment]
@YuWVandy what do you think?

**Comment by hzcheney** [comment]
Hi! @benedekrozemberczki Sorry about the late response, I have already finished the model part. There is a problem with the input feature named SSP(structural similarity profile), it consists of the drug similarity vector which is based on their fingerprint. The problem is I can't find a straightforward way to calculate the SSP, any idea?

**Comment by benedekrozemberczki** [comment]
It is the following:

1. For each drug a fingerprint is generated D X n. Where D is the number of drugs and n is the fingerprint dimensionality.
2. Using the fingerprints you define a D X D similarity matrix.
3. Using this Matrix you use PCA to reduce the dimensionality of the similarity matrix.
4. This would require on my side that we add a key to the dataset which we could use to retrieve the SSP vectors.

I would say using the drug feature vectors is sufficient to develop this.

**Comment by benedekrozemberczki** [comment]
Completely agree with @cthoyt about this. It should not be on the model side.

**Comment by benedekrozemberczki** [comment]
@hzcheney Are you going to open a PR with your code?

**Comment by hzcheney** [comment]
> @hzcheney Are you going to open a PR with your code?

@benedekrozemberczki Yeah! I have already opened a PR and please review it!
